nemo/collections/common/data/lhotse/nemo_adapters.py

```diff
@@ -290,31 +302,39 @@ def __iter__(self) -> Generator[Cut, None, None]:
                     )
                     data = shard_manifest[tar_info.name]
                     raw_audio = tar.extractfile(tar_info).read()
-                    # Note: Lhotse has a Recording.from_bytes() utility that we won't use here because
-                    #       the profiling indicated significant overhead in torchaudio ffmpeg integration
-                    #       that parses full audio instead of just reading the header for WAV files.
-                    # recording = lhotse.Recording.from_bytes(raw_audio, recording_id=tar_info.path)
-                    meta = soundfile.info(BytesIO(raw_audio))
-                    recording = Recording(
-                        id=tar_info.path,
-                        sources=[AudioSource(type="memory", channels=list(range(meta.channels)), source=raw_audio)],
-                        sampling_rate=int(meta.samplerate),
-                        num_samples=meta.frames,
-                        duration=meta.duration,
-                    )
-                    cut = recording.to_cut()
-                    cut.supervisions.append(
-                        SupervisionSegment(
-                            id=cut.id,
-                            recording_id=cut.recording_id,
-                            start=0,
-                            duration=cut.duration,
-                            text=data.get(self.text_field),
-                            language=data.get(self.lang_field),
-                        )
-                    )
-                    cut.custom = _to_custom_attr_dict(data)
-                    yield cut
+                    yield data, raw_audio, tar_info
+
+    def __iter__(self) -> Generator[Cut, None, None]:
```

Comment:
You will face conflicts when merging main because we extended this function with support for `tar` files with `offset` fields so that multiple manifest entries can refer to the same tar file. Let me know if you are unsure how to resolve them.